### PR TITLE
feat(editor/substation): read-only editor for SubEquipment element

### LINKED ISSUE #960: Add read-only SubEquipment editor to the Substation plugin
As a user of OpenSCD I want to see `SubEquipment` element in the SCL in the substation plugin

**Requirements**
- make use of `action-pane`
- title is "SubEquipment.name — SubEquipment.desc (SubEquipment.phase)"
- show in all possible parents that is `PowerTransformer`, `ConductingEquipment`, `TransformerWinding`, `TapChanger`
- show all possible children, that is: `LNode`, `EqFunction`
- show only with function filter disabled `showfunction=true`

### PR DESCRIPTION
**Pull request by pascalwilbrink**
closes #960

### CODE REVIEW

File: test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 /* @web/test-runner snapshot v1 */
 export const snapshots = {};
 
-snapshots["conducting-equipment-editor rendered as action icon looks like the latest snapshot"] = 
-`<action-icon
+snapshots[
```
Comment (dlabordus) [inline test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js:4]: Don't understand why these kind of formatting changes are made.

File: src/editors/substation/sub-equipment-editor.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  doc!: XMLDocument;
+  /** SCL element SubEquipment */
+  @property({ attribute: false })
+  element!: Element;
+
+  /** SubEquipment name attribute */
+  @property({ type: String })
+  get label(): string {
+    return `${this.element.getAttribute('name') ?? 'UNDEFINED'} - ${
+      this.element.getAttribute('desc') ?? 'UNDEFINED'
```
[the remark this answers is not in the document]
Comment (dlabordus) [reply src/editors/substation/sub-equipment-editor.ts:35]: Maybe also create different snapshot tests for this to show what happens when these attributes are empty.

**Review by dlabordus — CHANGES_REQUESTED** [review]

File: test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js (hunk 1 above)
Comment (pascalwilbrink) [reply test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js:4]: Fixed

File: src/editors/substation/sub-equipment-editor.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    } (${this.element.getAttribute('phase') ?? 'UNDEFINED'})`;
+  }
+
+  render(): TemplateResult {
```
Comment (JakobVogelsang) [inline src/editors/substation/sub-equipment-editor.ts:39]: Methods that are used in another methods are usuaelly on top of each other. Here
```
...renderEqFunction

...renderLNodes

...render
```

File: src/foundation.ts [hunk 4]
```diff
@@ -1699,7 +1699,11 @@ export const tags: Record<
     identity: namingIdentity,
     selector: namingSelector,
     parents: ['Process', 'Line', 'SubFunction', 'Function', 'Bay'],
-    children: [...tAbstractConductingEquipmentSequence, 'EqFunction'],
+    children: [
+      ...tAbstractConductingEquipmentSequence,
+      'EqFunction',
+      'SubEquipment',
```
Comment (JakobVogelsang) [inline src/foundation.ts:1705]: Thanks we missed that!

File: src/foundation.ts [hunk 5]
```diff
@@ -2127,6 +2131,7 @@ export const tags: Record<
       'TransformerWinding',
       'SubEquipment',
       'EqFunction',
+      'SubEquipment',
```
Comment (JakobVogelsang) [inline src/foundation.ts:2134]: Thanks we missed that!

File: src/foundation.ts [hunk 6]
```diff
@@ -2357,6 +2362,8 @@ export const tags: Record<
     parents: [
       'TapChanger',
       'PowerTransformer',
+      'ConductingEquipment',
```
Comment (JakobVogelsang) [inline src/foundation.ts:2365]: Thanks we missed that!

File: test/testfiles/valid2007B4.scd [hunk 7]
```diff
@@ -11,13 +11,27 @@
 	<Substation name="AA1" desc="Substation">
 		<VoltageLevel name="E1" desc="Voltage Level" nomFreq="50.0" numPhases="3">
 			<Voltage unit="V" multiplier="k">110.0</Voltage>
+			<PowerTransformer xmlns="http://www.iec.ch/61850/2003/SCL" name="pTrans" type="PTR">
```
Comment (JakobVogelsang) [inline test/testfiles/valid2007B4.scd:14]: I would not mess around in already existing test files too much. Here for example you cannot guarantee that it is still schema valid. I would work with the test file I gave you.

File: test/testfiles/valid2007B4withSubstationXY.scd [hunk 8]
```diff
@@ -204,7 +209,7 @@
 						</GSEControl>
 						<GSEControl type="GOOSE" appID="0003" fixedOffs="false" confRev="1" name="GCB2"/>
 					</LN0>
-					<LN lnClass="XCBR" inst="1" lnType="Dummy.XCBR1">
```
Comment (JakobVogelsang) [inline test/testfiles/valid2007B4withSubstationXY.scd:207]: Same here!

File: test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js (hunk 1 above)
Comment (JakobVogelsang) [reply test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js:4]: No it is not. You system is makeing `"` to `'`. When I have to touch that file it is again indicated as a change. There should not be any diff in this snapshot

File: test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js [hunk 9]
```diff
@@ -36,8 +37,9 @@ snapshots["conducting-equipment-editor rendered as action icon looks like the la
 `;
 /* end snapshot conducting-equipment-editor rendered as action icon looks like the latest snapshot */
 
-snapshots["conducting-equipment-editor rendered as action pane looks like the latest snapshot"] =
```
Comment (JakobVogelsang) [inline test/unit/editors/substation/__snapshots__/conducting-equipment-editor.test.snap.js:39]: Same here

File: test/unit/editors/substation/sub-equipment-editor.test.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+      await fixture(
+        html`<sub-equipment-editor
+          .element=${doc.querySelector('SubEquipment')}
+        ></sub-equipment-editor>`
+      )
+    );
+  });
+
+  it('looks like the latest snapshot', async () => {
+    await expect(element).shadowDom.to.equalSnapshot();
```
Comment (JakobVogelsang) [inline test/unit/editors/substation/sub-equipment-editor.test.ts:24]: There are test missing. How do I see that all the valid child elements are rendered as well when they are in the file.

**Review by JakobVogelsang — CHANGES_REQUESTED** [review]

File: test/unit/editors/substation/__snapshots__/voltage-level-editor.test.snap.js [hunk 11]
```diff
@@ -99,6 +99,10 @@ snapshots["voltage-level-editor looks like the latest snapshot"] =
       </mwc-list-item>
     </mwc-menu>
   </abbr>
+  <div class="actionicon ptrContent">
```
Comment (JakobVogelsang) [inline test/unit/editors/substation/__snapshots__/voltage-level-editor.test.snap.js:102]: Where is this coming from? Changed test file?

**Review by JakobVogelsang — COMMENTED** [review]
I cannot explain myself one snapshot. Other than that LGTM

**Review by JakobVogelsang — APPROVED** [review]
LGTM
